numcodecs/tests/test_registry.py

```diff
@@ -34,8 +34,8 @@ def test_all_classes_registered():
         if (
             inspect.isclass(obj)
             and issubclass(obj, numcodecs.abc.Codec)
+            and hasattr(obj, 'codec_id')
```

Comment:
```suggestion
            and obj is not numcodecs.abc.Codec
```
I think is a bit clearer in intent (we don't want or need the base codec to be registered)?